core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -137,16 +137,26 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
     * Appends `s2` to the end of this stream.
     *
     * @example {{{
-    * scala> ( Stream(1,2,3)++Stream(4,5,6) ).toList
+    * scala> (Stream(1,2,3) ++ Stream(4,5,6)).toList
     * res0: List[Int] = List(1, 2, 3, 4, 5, 6)
     * }}}
     *
-    * If `this` stream is not terminating, then the result is equivalent to `this`.
+    * If `this` stream is infinite, then the result is equivalent to `this`.
+    *
+    * Resources acquired in `this` stream are guaranteed to be finalized before `s2` is evaluated.
     */
-  def ++[F2[x] >: F[x], O2 >: O](s2: => Stream[F2, O2]): Stream[F2, O2] = append(s2)
+  def ++[F2[x] >: F[x], O2 >: O](s2: => Stream[F2, O2]): Stream[F2, O2] = scope.appendNoScope(s2)
 
   /** Appends `s2` to the end of this stream. Alias for `s1 ++ s2`. */
   def append[F2[x] >: F[x], O2 >: O](s2: => Stream[F2, O2]): Stream[F2, O2] =
+    this ++ s2
+
+  /**
+    * Like `append`/`++` but without the guarantee that resources acquired in this stream are
```

Comment:
> Would there be a way to do the latter behavior at all?

Can you think of a use case that actually benefits from those semantics? If not, I'd rather not publicise `appendNoScope` too much, I'm afraid that it will bite people in the same way the current behaviour is